gcloud/test_connection.py

```diff
@@ -330,10 +330,35 @@ def test_api_request_w_500(self):
         conn = self._makeMockOne()
         conn._http = _Http(
             {'status': '500', 'content-type': 'text/plain'},
-            '{}',
+            b'{}',
         )
         self.assertRaises(InternalServerError, conn.api_request, 'GET', '/')
 
+    def test_api_request_non_binary_response(self):
+        conn = self._makeMockOne()
+        http = conn._http = _Http(
+            {'status': '200', 'content-type': 'application/json'},
+            u'{}',
```

Comment:
I just wanted to preserve the majority of the code in #724.

Removing it leads to a branch miss in `Connection.api_request`:

```python
            if isinstance(content, six.binary_type): 
                content = content.decode('utf-8') 
            return json.loads(content)
```

I thought this approach was friendlier than raising if the response was unicode.